### PR TITLE
[flutter_tools] remove flutter view cache

### LINKED ISSUE #56194: Error connecting to the service protocol: No Flutter view is available on "device_name"
## Steps to Reproduce

1. Trying to run flutter app after recent update in master channel with `flutter run -v`
2. It installs the APK but gets disconnected with the following error message.
`Error connecting to the service protocol: No Flutter view is available on "device_name"`

`flutter doctor` output:
```
[√] Flutter (Channel master, 1.18.0-9.0.pre.93, on Microsoft Windows [Version 10.0.18362.720], locale en-IN)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Chrome - develop for the web
[√] Android Studio (version 3.5)
[√] VS Code (version 1.44.2)
[√] Connected device (3 available)

• No issues found!
```
## Logs

```
[ +190 ms] Successfully connected to service protocol: http://127.0.0.1:54365/RAoQ3a4zxzY=/
[   +7 ms] Error connecting to the service protocol: No Flutter view is available on ONEPLUS A6000.
[   +4 ms] "flutter run" took 57,058ms.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
#1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:570:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:796:18)
#3      _rootRunUnary (dart:async/zone.dart:1192:38)
#4      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#5      _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#7      Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#8      Future._completeWithValue (dart:async/future_impl.dart:526:5)
#9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
#10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:303:13)
#11     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#12     _rootRunUnary (dart:async/zone.dart:1192:38)
#13     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#14     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#15     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#16     Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#17     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#18     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
#19     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:303:13)
#20     AndroidDevice.isLocalEmulator (package:flutter_tools/src/android/android_device.dart)
#21     _rootRunUnary (dart:async/zone.dart:1192:38)
#22     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#23     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#24     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:686:45)
#25     Future._propagateToListeners (dart:async/future_impl.dart:715:32)
#26     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#27     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:560:7)
#28     _rootRun (dart:async/zone.dart:1184:13)
#29     _CustomZone.run (dart:async/zone.dart:1077:19)
#30     _CustomZone.runGuarded (dart:async/zone.dart:979:7)
#31     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019:23)
#32     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#33     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#34     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#35     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)

```

```
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fixes https://github.com/flutter/flutter/issues/56194

Remove caching of FlutterView. Perhaps the FlutterView RPC might return an empty list if the VM is not quite up yet? We had some old logic to poll the flutter views RPC for up to 200ms. That doesn't seem like a great approach, so instead we could forgo it entirely and trust that either the views come up before the developer tries to interact, or we crash.

https://github.com/flutter/flutter/blob/stable/packages/flutter_tools/lib/src/vmservice.dart#L1070

### CODE REVIEW

**Comment by jonahwilliams** [comment]
nvm, i misread the code, we waited forever lol. I wonder if instead we should wait for isolate runnable? Otherwise I could patch wait forever back in, seems bad to do polling if we don't need to.

**Comment by zanderso** [comment]
I think I don't quite follow, but waiting forever here sounds in line with the policy against timeouts.

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/5437 was pretty unhelpful in trying to figure out why the cache was added in the first place.

**Comment by jonahwilliams** [comment]
I added back the poll forever logic

File: packages/flutter_tools/lib/src/vmservice.dart [hunk 1]
```diff
@@ -730,15 +730,29 @@ extension FlutterVmService on vm_service.VmService {
...
+      );
+      final List<Object> rawViews = response.json['views'] as List<Object>;
+      final List<FlutterView> views = <FlutterView>[
+        for (final Object rawView in rawViews)
+          FlutterView.parse(rawView as Map<String, Object>)
+      ];
+      if (views.isNotEmpty || returnEarly) {
+        return views;
+      }
+      await Future<void>.delayed(delay);
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/vmservice.dart:754]: This previously had a 'This is taking a long time' message. Should that also be restored, or is this already wrapped in a status spinner thing?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/vmservice.dart:754]: Ahh, status spinner is a good idea. Let me add that instead of adding a print

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/vmservice.dart:754]: Done

**Comment by jonahwilliams** [comment]
Status spinner added, tests updated - PTAL

**Comment by jonahwilliams** [comment]
The Linux failure is a flake that I cannot rerun

**Review by jmagman — APPROVED** [review]
LGTM, let's see if the attach test goes green.

**Pull request merged**